### PR TITLE
MINOR: [R] [Docs] Fix link in ?cast documentation

### PR DESCRIPTION
**Pull request by amoeba**

### CODE REVIEW

File: r/R/dplyr-funcs-type.R [hunk 1]
```diff
@@ -43,7 +43,7 @@ register_bindings_type <- function() {
 #'   mutate(cyl = cast(cyl, string()))
 #' }
 #' @keywords internal
-#' @seealso https://arrow.apache.org/docs/cpp/api/compute.html for the list of
+#' @seealso [Arrow C++ documentation](https://arrow.apache.org/docs/cpp/compute.html) for the list of
```
[the remark this answers is not in the document]
Comment (amoeba) [reply r/R/dplyr-funcs-type.R:46]: Changing the link was actually a mistake on my part and I think linking to the cast options in the API docs makes the most sense. Not sure on which link is more stable but I'll note that https://arrow.apache.org/docs/cpp/api/compute.html?highlight=castoptions#arrow%3A%3Acompute%3A%3ACastOptions also works and is maybe preferable?

Re:

> - manually describe the options here, and update them if new options are implemented - perhaps the API is stable enough now that this is unlikely

At first glance, I wonder if this is too much detail. Do users tend to use CastOptions much from R or do we expect them to? Maybe a separate page would be better or we could add more info to `?cast_options`? I could go either way (just link to the C++ docs from ?cast versus fill this in more, as you suggest).

[the remark this answers is not in the document]
Comment (thisisnic) [reply r/R/dplyr-funcs-type.R]: > Do users tend to use CastOptions much from R or do we expect them to?

I don't have a good answer for you for this, but in the absence of evidence to suggest they are, I'll go with "no".  I like the link that you found and the changes you made here, so let's go with those - we can always come back to this if there is user need for it.

**Review by thisisnic — APPROVED** [review]
Wonderful, always good to see these PRs which make things easier for our users to find the information they need! Thanks!

File: r/R/dplyr-funcs-type.R [hunk 2]
```diff
@@ -43,8 +43,9 @@ register_bindings_type <- function() {
 #'   mutate(cyl = cast(cyl, string()))
 #' }
 #' @keywords internal
-#' @seealso https://arrow.apache.org/docs/cpp/api/compute.html for the list of
-#' supported CastOptions.
+#' @seealso [`data-type`] for a list of [DataType] to be used with `to`.
+#' @seealso [Arrow C++ CastOptions documentation](https://arrow.apache.org/docs/cpp/api/compute.html?highlight=castoptions#arrow%3A%3Acompute%3A%3ACastOptions)
```
Comment (thisisnic) [inline r/R/dplyr-funcs-type.R:47]:
```suggestion
#' @seealso [Arrow C++ CastOptions documentation](https://arrow.apache.org/docs/cpp/api/compute.html?highlight=castoptions#arrow%3A%3Acompute%3A%3ACastOptions) # nolint
```

File: r/R/dplyr-funcs-type.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (nealrichardson) [reply r/R/dplyr-funcs-type.R]: Ironically, we don't actually support all of the cast options yet: ARROW-17605 (should be a straightforward change if someone wanted to pick it up)

**Pull request merged**